libs/api/ebpf_api.cpp

```diff
@@ -2224,6 +2231,9 @@ _initialize_ebpf_object_from_native_file(
         program->attach_type = info->expected_attach_type;
         program->fd = ebpf_fd_invalid;
 
+        // Only autoload programs with a defined program type.
+        program->autoload = memcmp(&program->program_type, &empty_program_type, sizeof(empty_program_type)) != 0;
```

Comment:
nit: will it ever happen for a native module that program type for a program is empty? I think the program type will always be present in the native module, else the verification would not have succeeded.

Reply:
Not sure.  Certainly for an ELF file you can have one that just has maps and no programs in it.  So I want the same logic here in case the same can happen for a .sys file now or in the future.

Reply:
Verification succeeds if there are 0 programs in the file.

Reply:
Few points:

1. bpf2c today returns error if there are no program sections found in the ELF file.
2. ebpfcore also fails to load the native module if there are 0 programs in the native module.

Reply:
If so, I think both of those are bugs.

Reply:
Fixed ebpfcore.   Left bpf2c for a separate PR.